packages/react-button/src/components/Button/useButtonBehavior.ts

```diff
@@ -0,0 +1,16 @@
+import { buttonBehavior } from '@fluentui/accessibility';
+import { useBehaviorKeyActions } from '@uifabric/react-hooks';
+import { ButtonProps, ButtonState } from './Button.types';
+
+export const useButtonBehavior = (props: ButtonProps, ref: React.Ref<HTMLElement>): Partial<ButtonState> => {
+  const { as, disabled, loading, onClick, onKeyDown: _onKeyDown } = props;
+
+  const { attributes, keyActions } = buttonBehavior({
+    as: (ref !== null && typeof ref !== 'function' && ref.current?.tagName) || (as as string),
+    disabled,
+    loading,
+  });
```

Comment:
This will not work 😭 

If you will check the original suggestion it works based on `elementType` that is stored under the state (https://github.com/reakit/reakit/blob/9bd6882fb50948c616e757d750b436decc36ddfc/packages/reakit/src/Button/Button.ts#L40). I made a simplified CSB to highlight it: https://codesandbox.io/s/thirsty-galois-tdz7s?file=/src/App.js

`ref.current?.tagName` will be nothing during the first render as refs are resolved during mount stage.

---

```suggestion
  const [elementTag, setElementTag] = React.useState("button");

  const { attributes, keyActions } = buttonBehavior({
    as: tagName,
    disabled,
    loading,
  });
  
    React.useEffect(() => {
      const element = ref.current;

      if (!element) {
        console.warn(
          "Can't determine whether the element is a native button because `ref` wasn't passed to the component",
        );
        return;
      }

      setElementTag(element.tagName.toLowerCase())
    }, []);
```

Reply:
Changed!